### PR TITLE
schemas: splits photos into two schemas

### PR DESCRIPTION
**Pull request by sanderpick**
`camera_roll` for storage, `media` for photos (eventually other types that belong in a shared thread / chat / convo. For this to be app ready, I need to enable taking an existing hash (the blob file from the `camera_roll` schema, and using it as the `:file` tag in the `media` schema... all the handling is already there (it's the same as passing one step to another), but need a little tweak for directly setting the input.

### CODE REVIEW

File: cmd/threads.go [hunk 1]
```diff
@@ -51,11 +51,12 @@ purposes and 1-to-1 communication channels.
...
-	Key    string         `short:"k" long:"key" description:"A locally unique key used by an app to identify this thread on recovery."`
-	Open   bool           `short:"o" long:"open" description:"Set the thread type to open (default private)."`
-	Schema flags.Filename `short:"s" long:"schema" description:"Thread Schema filename. Superseded by built-in schema flags."`
-	Photos bool           `long:"photos" description:"Use the built-in photo Schema."`
+	Client     ClientOptions  `group:"Client Options"`
+	Key        string         `short:"k" long:"key" description:"A locally unique key used by an app to identify this thread on recovery."`
+	Open       bool           `short:"o" long:"open" description:"Set the thread type to open (default private)."`
+	Schema     flags.Filename `short:"s" long:"schema" description:"Thread Schema filename. Supersedes the built-in schema flags."`
+	Media      bool           `long:"media" description:"Use the built-in media Schema."`
+	CameraRoll bool           `long:"camera-roll" description:"Use the built-in camera roll Schema."`
```
[the remark this answers is not in the document]
Comment (asutula) [reply cmd/threads.go]: Yea, sounds like an improvement we can make, but like I said, not critical for a first cut.

File: schema/textile/media.go [hunk 2]
```diff
@@ -1,49 +1,33 @@
...
+      "use": ":file",
       "mill": "/image/resize",
       "opts": {
         "width": "800",
         "quality": "80"
       }
     },
     "small": {
-      "use": "raw",
+      "use": "large",
```
Comment (carsonfarmer) [inline schema/textile/media.go]: nice, like it

File: schema/textile/camera_roll.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    "exif": {
+      "use": "raw",
+      "mill": "/image/exif"
+    },
+    "thumb": {
+      "use": "raw",
+      "pin": true,
+      "mill": "/image/resize",
+      "opts": {
+        "width": "100",
```
[the remark this answers is not in the document]
Comment (sanderpick) [reply schema/textile/camera_roll.go]: ok cool

[the remark this answers is not in the document]
Comment (sanderpick) [reply schema/textile/camera_roll.go]: upped it to 320 like the media "small"

**Comment by sanderpick** [comment]
Ok, this should be good to go. When preparing files, you can pass an existing file hash as the file "path", which is what we'll want to do when sharing into a shared thread from camera roll.

**Comment by carsonfarmer** [comment]
Reran pipeline, should be good to merge now 👍

**Comment by sanderpick** [comment]
Changes the schemas slightly to ensure that `media`'s small size is the same as `camera_roll`'s `thumb` size, which results in less more de-duplication.

**Pull request merged**